drivers/python/rethinkdb/net_twisted.py

```diff
@@ -224,40 +182,47 @@ def _empty_error(self):
     @inlineCallbacks
     def fetch_next(self, wait=True):
         timeout = Cursor._wait_to_timeout(wait)
-        start_timeout(timeout, self.new_response)
```

Comment:
Since this may involve multiple waits, a relative timeout isn't useful, we need to know the deadline and adjust the timeout each time we need to wait.  In addition, multiple places may have called this function, so we can't all use the same `Deferred` object and chain timeouts onto it.  For this reason, the cursor now contains a list of waiters (`Deferred`s), each of which may have its own timeout.